sw/device/lib/testing/test_rom/test_rom.c

```diff
@@ -80,6 +82,13 @@ bool rom_test_main(void) {
   // Print the chip version information
   LOG_INFO("%s", chip_info);
 
+  // Skip sram_init for test_rom
+  uint32_t reset_reasons = rstmgr_reason_get();
```

Comment:
any reason you use the mask rom driver code here instead of the DIF `dif_rstmgr_reset_info_get()`? This driver code seems like it also collects the crashdump info into a global (https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/drivers/rstmgr.c;drc=94cdceaa94c19ce1876c03ef7d4c19f69c887c0a;l=73) which might be uncessary?

Reply:
hmm thats odd. Did you make sure to call `dif_rstmgr_init()` to initialize the handle before calling `dif_rstmgr_reset_info_get()`?

Reply:
I did not... I'll try that and check all the tests still pass, and then update the EB fakes to cover the functions in the DIF as well.

Reply:
It works but will need some more work/indirection to fake the rstmgr DIF for EB.

Reply:
#13642 Shows my current thinking on what needs to be done to fake the relevant functions in the rstmgr DIF

Reply:
since this is handled in a separate PR, can I merge this one?

Reply:
Yeah, that should be good. Thanks!